packages/moonstone/MoonstoneDecorator/MoonstoneDecorator.js

```diff
@@ -118,17 +120,113 @@ const MoonstoneDecorator = hoc(defaultConfig, (config, Wrapped) => {
 	const Decorator = class extends React.Component {
 		static displayName = 'MoonstoneDecorator';
 
+		static contextTypes = contextTypes
+
+		static childContextTypes = contextTypes
+
+		static propTypes =  /** @lends moonstone/MoonstoneDecorator.AccessibilityDecorator.prototype */ {
+			/**
+			 * Enables additional features to help users visually differentiate components.
+			 * The UI library will be responsible for using this information to adjust
+			 * the components' contrast to this preset.
+			 *
+			 * @type {Boolean}
+			 * @public
+			 */
+			highContrast: PropTypes.bool,
+
+			/**
+			 * Set the goal size of the text. The UI library will be responsible for using this
+			 * information to adjust the components' text sizes to this preset.
+			 * Current presets are `'normal'` (default), and `'large'`.
+			 *
+			 * @type {String}
+			 * @default 'normal'
+			 * @public
+			 */
+			textSize: PropTypes.oneOf(['normal', 'large'])
+		}
+
+		static defaultProps = {
+			highContrast: false,
+			textSize: 'normal'
+		}
+
+		getChildContext () {
+			return {
+				Subscriber: this.publisher.getSubscriber()
+			};
+		}
+
+		componentWillMount () {
+			if (textSize || highContrast) {
+				this.publisher = Publisher.create('resize', this.context.Subscriber);
+			}
+
+			if (typeof window === 'object') {
+				const palmSystem = window.PalmSystem;
+
+				Spotlight.initialize({
+					selector: '.' + spottableClass,
+					restrict: 'none'
+				});
+
+				Spotlight.set(rootContainerId, {
+					overflow: true
+				});
+
+				if (palmSystem && palmSystem.cursor) {
+					Spotlight.setPointerMode(palmSystem.cursor.visibility);
+				}
+			}
+		}
+
+		componentDidMount () {
+			if (!noAutoFocus) {
+				Spotlight.focus();
+			}
+			if (config.dynamic) window.addEventListener('resize', this.handleResize);
+			// eslint-disable-next-line react/no-find-dom-node
+			// this.rootNode = ReactDOM.findDOMNode(this);
+		}
+
+		componentDidUpdate (prevProps) {
+			if (prevProps.textSize !== this.props.textSize) {
+				this.publisher.publish({
```

Comment:
If the user decides to opt-out for textSize, there won't be a publisher created.